x/incentive/types/msg.go

```diff
@@ -39,35 +33,33 @@ func (msg MsgClaimUSDXMintingReward) Type() string { return "claim_usdx_minting_
 
 // ValidateBasic does a simple validation check that doesn't require access to state.
 func (msg MsgClaimUSDXMintingReward) ValidateBasic() error {
-	if msg.Sender.Empty() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "sender address cannot be empty")
+	_, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "sender address cannot be empty or invalid")
 	}
-	if err := MultiplierName(msg.MultiplierName).IsValid(); err != nil {
+	if err := NewMultiplierNameFromString(msg.MultiplierName).IsValid(); err != nil {
 		return err
 	}
 	return nil
 }
 
 // GetSignBytes gets the canonical byte representation of the Msg.
 func (msg MsgClaimUSDXMintingReward) GetSignBytes() []byte {
-	bz := ModuleCdc.MustMarshalJSON(msg)
+	bz := ModuleCdc.MustMarshalJSON(&msg)
 	return sdk.MustSortJSON(bz)
 }
 
 // GetSigners returns the addresses of signers that must sign.
 func (msg MsgClaimUSDXMintingReward) GetSigners() []sdk.AccAddress {
-	return []sdk.AccAddress{msg.Sender}
-}
-
-// MsgClaimUSDXMintingRewardVVesting message type used to claim USDX minting rewards for validator vesting accounts
-type MsgClaimUSDXMintingRewardVVesting struct {
-	Sender         sdk.AccAddress `json:"sender" yaml:"sender"`
-	Receiver       sdk.AccAddress `json:"receiver" yaml:"receiver"`
-	MultiplierName string         `json:"multiplier_name" yaml:"multiplier_name"`
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{sender}
 }
 
 // NewMsgClaimUSDXMintingRewardVVesting returns a new MsgClaimUSDXMintingReward.
-func NewMsgClaimUSDXMintingRewardVVesting(sender, receiver sdk.AccAddress, multiplierName string) MsgClaimUSDXMintingRewardVVesting {
+func NewMsgClaimUSDXMintingRewardVVesting(sender, receiver string, multiplierName string) MsgClaimUSDXMintingRewardVVesting {
```

Comment:
For the `NewMsgxxx()` functions I've been keeping the address args as `sdk.AccAddress` then converting it to string in the returned struct, eg

```go
return MsgClaimUSDXMintingRewardVVesting{
		Sender:         sender.String(),
                Receiver:       receiver.String(),
		MultiplierName: multiplierName,
	}
}
```

Depends on where the functions are used, but seems like they're mostly when we already have an address so it might be easier for use that way and prevents possible need to revalidate here